src/octoprint/util/__init__.py

```diff
@@ -1078,27 +1078,11 @@ def __exit__(self):
             self.cleanup()
 
 
-def bom_aware_open(filename, encoding="ascii", mode="r", **kwargs):
-    import codecs
-
+def bom_aware_open(filename, encoding="utf-8", mode="r", **kwargs):
     assert "b" not in mode, "binary mode not support by bom_aware_open"
 
-    codec = codecs.lookup(encoding)
-    encoding = codec.name
-
-    if kwargs is None:
-        kwargs = {}
-
-    potential_bom_attribute = "BOM_" + codec.name.replace("utf-", "utf").upper()
-    if "r" in mode and hasattr(codecs, potential_bom_attribute):
-        # these encodings might have a BOM, so let's see if there is one
-        bom = getattr(codecs, potential_bom_attribute)
-
-        with io.open(filename, mode="rb") as f:
-            header = f.read(4)
-
-        if header.startswith(bom):
-            encoding += "-sig"
+    if encoding in {"utf-8", "utf8"} and "r" in mode:
+        encoding = "utf-8-sig"
```

Comment:
This turns this into a breaking change. The old code will work with any encoding that supports a BOM (as retrieved from `codecs`), this code here only works with `utf8`. While it is extremely unlikely that there is client code out there using anything but UTF8 with `bom_aware_open`, we can't rule it out and thus this change would go against semver.

Suggestion:

```suggestion
    import codecs
    codec = codecs.lookup(encoding)
    potential_bom_attribute = "BOM_" + codec.name.replace("utf-", "utf").upper()
    if hasattr(codecs, potential_bom_attribute):
        encoding = codec.name + "-sig"
```

Reply:
This isn't quite right, because only utf-8 supports the `-sig`. [UTF-16 is required to handle a BOM by default, and so is UTF-32, although it isn't typically used.](https://en.wikipedia.org/wiki/Byte_order_mark#Usage)